src/Shared/TaskLoggingHelper.cs

```diff
@@ -235,9 +235,19 @@ public virtual string GetResourceMessage(string resourceName)
             string resourceString = FormatResourceString(resourceName, null);
             return resourceString;
         }
-#endregion
+        #endregion
+
+        #region Message logging methods
 
-#region Message logging methods
+        /// <summary>
+        /// Returns true if a message of given importance should be logged because it is possible that a logger consuming it exists.
+        /// </summary>
+        /// <param name="importance">The importance to check.</param>
+        /// <returns>True if messages of the given importance should be logged, false if it's guaranteed that such messages would be ignored.</returns>
+        public bool ShouldLogMessage(MessageImportance importance)
+        {
+            return importance <= ((IBuildEngine10)BuildEngine).EngineInterface.MinimumRequiredMessageImportance;
```

Comment:
Verify (maybe via reflection?) that it exists?I don't think it's critical in this case, but it would be necessary for any later additions to EngineInterface.

Reply:
This I'm not sure is necessary. In the previous model trying to cast to `IBuildEngineN` would never gracefully fail, as the type would fail to bind at JIT time should MSBuild be older than what the caller was built against.

In other words, I don't think "light-up" should be the goal here because it never worked.

Reply:
As discussed, light-up **is** a goal and will be implemented by exposing a version prop.